docs/dev/table/catalog.md

```diff
@@ -23,344 +23,151 @@ specific language governing permissions and limitations
-
-- hive-exec-2.3.4.jar // contains hive-metastore-2.3.4
-
-
-// Hadoop dependencies
-- flink-shaded-hadoop-2-uber-2.7.5-1.8.0.jar
-- flink-hadoop-compatibility-{{site.version}}.jar
-
-```
-
-For Hive 1.2.1, users need:
-
-```
-// Hive dependencies
-
-- hive-metastore-1.2.1.jar
-- hive-exec-1.2.1.jar
-- libfb303-0.9.3.jar
-
-
-// Hadoop dependencies
-- flink-shaded-hadoop-2-uber-2.6.5-1.8.0.jar
-- flink-hadoop-compatibility-{{site.version}}.jar
-
-```
-
-If you don't have Hive dependencies at hand, they can be found at [mvnrepostory.com](https://mvnrepository.com):
-
-- [hive-exec](https://mvnrepository.com/artifact/org.apache.hive/hive-exec)
-- [hive-metastore](https://mvnrepository.com/artifact/org.apache.hive/hive-metastore)
-
-Note that users need to make sure the compatibility between their Hive versions and Hadoop versions. Otherwise, there may be potential problem, for example, Hive 2.3.4 is compiled against Hadoop 2.7.2, you may run into problems when using Hive 2.3.4 with Hadoop 2.4.
-
-
-### Data Type Mapping
-
-For both Flink and Hive tables, `HiveCatalog` stores table schemas by mapping them to Hive table schemas with Hive data types. Types are dynamically mapped back on read.
-
-Currently `HiveCatalog` supports most Flink data types with the following mapping:
-
-|  Flink Data Type  |  Hive Data Type  |
-|---|---|
-| CHAR(p)       |  CHAR(p)* |
-| VARCHAR(p)    |  VARCHAR(p)** |
-| STRING        |  STRING |
-| BOOLEAN       |  BOOLEAN |
-| TINYINT       |  TINYINT |
-| SMALLINT      |  SMALLINT |
-| INT           |  INT |
-| BIGINT        |  LONG |
-| FLOAT         |  FLOAT |
-| DOUBLE        |  DOUBLE |
-| DECIMAL(p, s) |  DECIMAL(p, s) |
-| DATE          |  DATE |
-| TIMESTAMP_WITHOUT_TIME_ZONE |  TIMESTAMP |
-| TIMESTAMP_WITH_TIME_ZONE |  N/A |
-| TIMESTAMP_WITH_LOCAL_TIME_ZONE |  N/A |
-| INTERVAL      |   N/A*** |
-| BINARY        |   N/A |
-| VARBINARY(p)  |   N/A |
-| BYTES         |   BINARY |
-| ARRAY\<E>     |  ARRAY\<E> |
-| MAP<K, V>     |  MAP<K, V> ****|
-| ROW           |  STRUCT |
-| MULTISET      |  N/A |
-
-
-Note that we only cover most commonly used data types for now.
-
-The following limitations in Hive's data types impact the mapping between Flink and Hive:
-
-\* maximum length is 255
-
-\** maximum length is 65535
-
-\*** `INTERVAL` type can not be mapped to hive `INTERVAL` for now.
-
-\**** Hive map key type only allows primitive types, while Flink map key can be any data type.
-
-## User-configured Catalog
-
-Catalogs are pluggable. Users can develop custom catalogs by implementing the `Catalog` interface, which defines a set of APIs for reading and writing catalog meta-objects such as database, tables, partitions, views, and functions.
-
-Catalog Registration
---------------------
-
-## Register Catalog in Table API
-
-To register a catalog in Table API, users can create a catalog instance and register it through `TableEnvironment.registerCatalog(name, catalog)`.
-
-## Register Catalog in SQL CLI
```

Comment:
we should keep this section somewhere, its last two paragraphs are very important information

Reply:
Done, moved to the user-defined catalog section